website/client/js/services/userServices.js

```diff
@@ -103,8 +104,28 @@ angular.module('habitrpg')
 
           return Tasks.getUserTasks();
         })
+        // refresh all but completed todos
         .then(function (response) {
-          var tasks = response.data.data;
+          tasks = response.data.data;
+
+          // only refresh completed todos if the user has the completed tabs list open
+          var taskButton = document.getElementById('completedTasksButton');
```

Comment:
was able to get around this by also checking to make sure that the current page title was tasks. I'll be pushing the new revision here shortly.